src/main/java/com/google/step/servlets/HomePageServlet.java

```diff
@@ -0,0 +1,219 @@
+    Map<String, Object> homePageMap = new HashMap<>();
+    homePageMap.put("popularDeals", homePageDealsMaps.get(0));
+    homePageMap.put("usersIFollow", homePageDealsMaps.get(1));
+    homePageMap.put("restaurantsIFollow", homePageDealsMaps.get(2));
+    homePageMap.put("tagsIFollow", homePageDealsMaps.get(3));
+    response.getWriter().println(JsonFormatter.getHomePageJson(homePageMap));
+  }
+
+  /** Creates a list of list of deal maps for the home page */
+  private List<List<Map<String, Object>>> getHomePageDealMaps(List<List<Deal>> homePageDeals) {
+    List<List<Map<String, Object>>> homePageDealsMapList = new ArrayList<>();
+    for (List<Deal> dealList : homePageDeals) {
+      List<Map<String, Object>> homePageSectionDealMaps = new ArrayList<>();
+      for (Deal deal : dealList) {
+        User user = userManager.readUser(deal.posterId);
+        Restaurant restaurant = restaurantManager.readRestaurant(deal.restaurantId);
+        List<Tag> tags = tagManager.readTags(dealTagManager.getTagIdsOfDeal(deal.id));
+        int votes = voteManager.getVotes(deal.id);
+        Map<String, Object> homePageDealMap =
+            JsonFormatter.getBriefHomePageDealMap(deal, user, restaurant, tags, votes);
+        homePageSectionDealMaps.add(homePageDealMap);
+      }
+      homePageDealsMapList.add(homePageSectionDealMaps);
+    }
+    return homePageDealsMapList;
+  }
+
+  /** Retrieves deals posted by tags followed by user */
+  private List<Deal> getDealsPublishedByTags(Set<Long> tagIds) {
+    List<Long> dealIdResults = new ArrayList<>();
+    for (Long id : tagIds) {
+      List<Long> dealIdsWithTag = dealTagManager.getDealIdsWithTag(id);
+      dealIdResults.addAll(dealIdsWithTag);
+    }
+    // Get rid of duplicate dealID (Deals with multiple tags)
+    List<Long> dealsWithoutDuplicates = new ArrayList<>(new HashSet<>(dealIdResults));
+    return dealManager.readDeals(dealsWithoutDuplicates);
+  }
+
+  /** Sorts deals based on new (Newest to oldest) */
+  private List<Deal> sortDealsBasedOnNew(List<Deal> deals) {
+    Collections.sort(
+        deals,
+        new Comparator<Deal>() {
+          @Override
+          public int compare(Deal deal1, Deal deal2) {
+            return LocalDateTime.parse(deal2.creationTimeStamp)
+                .compareTo(LocalDateTime.parse(deal1.creationTimeStamp)); // Descending
+          }
+        });
+    return deals;
+  }
+
+  private double epochSeconds(String timestamp) {
+    LocalDateTime time = LocalDateTime.parse(timestamp);
+    long epoch = time.atZone(ZoneId.of(LOCATION)).toEpochSecond();
+    return epoch;
+  }
+
+  /**
+   * Calculates a hot score for each deal entity, which takes into account both the time and the
+   * amount of votes it got
+   */
+  private double calculateHotScore(Deal deal, int votes) {
+    double order = Math.log(Math.max(Math.abs(votes), 1));
+    int sign = 0;
+    if (votes > 0) sign = 1;
+    else if (votes < 0) sign = -1;
```

Comment:
```suggestion
    if (votes > 0) {
      sign = 1;
    } else if (votes < 0) {
      sign = -1;
    }
```

https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used